src/model.js

```diff
@@ -222,6 +222,16 @@ function queryArrayAt(i) {
   return r;
 }
 
+// Internal: Provides the implementation of the query array to empty and then refill its content and update its `meta`.
+function queryArrayReset(newInitialArray = [], meta) {
+  this.replace(newInitialArray);
+  this.meta = meta;
+  if (meta && meta.totalCount) {
+    this.length = meta.totalCount;
+  }
+  this.error = undefined;
+};
```

Comment:
We should return `this` here.

Reply:
for chaining?

Reply:
Yes, and for general consistency with the rest of the library.